pants

```diff
@@ -33,24 +31,58 @@ SETUPTOOLS_REQUIREMENT="setuptools==5.4.1"
 VENV_PACKAGE=virtualenv-${VENV_VERSION}
 VENV_TARBALL=${VENV_PACKAGE}.tar.gz
 
+COLOR_RED="\x1b[31m"
+COLOR_GREEN="\x1b[32m"
+COLOR_RESET="\x1b[0m"
+
+function log() {
+  echo -e "$@" 1>&2
+}
+
+function die() {
+  (($# > 0)) && log "${COLOR_RED}$*${COLOR_RESET}"
+  exit 1
+}
+
+function green() {
+  (($# > 0)) && log "${COLOR_GREEN}$*${COLOR_RESET}"
+}
+
+function tempdir {
+  mktemp -d "$1"/pants.XXXXXX
+}
+
+function get_pants_ini_config_value {
+  config_key="$1"
+  valid_delimiters="[:=]"
+  optional_space="[[:space:]]*"
+  prefix="^${config_key}${optional_space}${valid_delimiters}${optional_space}"
+  sed -ne "/${prefix}/ s#${prefix}##p" pants.ini
+}
+
 # The high-level flow:
-# 1.) Grab pants version from pants.ini or default to latest.
-# 2.) Check for a venv via a naming/path convention and execute if found.
-# 3.) Otherwise create venv and re-exec self.
+# 1.) Resolve the Python interpreter, defaulting to `python2.7`.
+# 2.) Grab pants version from pants.ini or default to latest.
+# 3.) Check for a venv via a naming/path convention and execute if found.
+# 4.) Otherwise create venv and re-exec self.
 #
 # After that pants itself will handle making sure any requested plugins
 # are installed and up to date.
 
-function tempdir {
-  mktemp -d "$1"/pants.XXXXXX
+function determine_pants_runtime_python_version {
+  python_bin_name="${PYTHON:-python2.7}"
```

Comment:
Since `PYTHON` remains a "parameter" to this entire script, having it near the top of the file like it was before seems like good documentation of that fact.